src/Core/ConstraintValidator/Constraints/Reduction.php

```diff
@@ -0,0 +1,50 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\ConstraintValidator\Constraints;
+
+use PrestaShop\PrestaShop\Core\ConstraintValidator\ReductionValidator;
+use Symfony\Component\Validator\Constraint;
+
+/**
+ * Provides reduction validation by reduction type
+ */
+final class Reduction extends Constraint
+{
+    public $invalidTypeMessage = 'Reduction type is invalid.';
+
+    public $invalidAmountMessage = 'Reduction value is invalid.';
+
+    public $invalidPercentageMessage = 'Reduction value is invalid.';
```

Comment:
Why do you need 3 different properties for messages? Aren't 2 enough: 
`$invalidTypeMessage` - when reduction type is invalid (for example type is not `percent` nor `amount`)
`$invalidValueMessage` - when reduction value is invalid (for example value is 110 when percent is type)

Reply:
But the cap of 100% should be applied only for percentage value, else its constrained to be >= 0. So the error messages differs a bit

Reply:
I updated the default messages in constraints, now it should make sense